reset_factor.php

```diff
@@ -27,27 +27,53 @@
 require_once($CFG->libdir . '/adminlib.php');
 admin_externalpage_setup('tool_mfa_resetfactor');
 
+if (!empty($SESSION->bulk_users)) {
+    $bulk = true;
+    $users = array_map(function($user) {
+        return \core_user::get_user($user);
```

Comment:
This would be very heavy on the memory in one big hit. How about if if you left this array as just the ids, and did the user record lookup incrementally in the loop below so you only held one user in memory at a time? The loop can figure out if it is given just an id or a full object